packages/nimble-components/src/combobox/specs/IxD.md

```diff
@@ -0,0 +1,241 @@
+
+### Anatomy
+
+![ ](spec-images/combobox-1.png)
+
+| Element         | Description                                                                 |
+| --------------- | --------------------------------------------------------------------------- |
+| Label           | Text that tells the end-user what to expect in the component                |
+| Field input     | Input that displays the current selected or entered option                  |
+| Dropdown button | Click area that opens the dropdown                                          |
+| Dropdown        | A list of options to choose from displayed as an _open_ state               |
+| List item       | A selection the end-user can make, shown with other options in the dropdown |
+
+#### Label
+
+Comboboxes should always have a label, unless approved by an interaction designer or user researcher (usually in toolbar use cases). _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Field Input
+
+Comboboxes can be empty by default or have a default selection from the menu. Empty comboboxes should include placeholder text that propts the end-user to enter a custom value or select an option from the menu. Placeholder text should always follow the pattern "Enter or select [thing(s)]", for example "Enter or select address". Ellipses are not needed for the placeholder. Use sentence casing for the placeholder.
+
+![ ](spec-images/combobox-2.png)
+
+#### Dropdown & List Items
+
+A list of options to select, displayed as an open state._See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+### Related Components
+
+To allow end-users to select a list item but not enter a custom value, use a [**select**](/packages/nimble-components/src/select/specs/IxD.md).
+
+![ ](spec-images/combobox-alternative-select.png)
+
+To allow end-users to multi-select list items, use a [**tag picker**](/packages/nimble-components/src/tag-picker/specs/IxD.md).
+
+![ ](spec-images/combobox-alternative-picker.png)
+
+To allow end-users to select less than three options, use a [**radio group**](/packages/nimble-components/src/radio-group/specs/IxD.md).
+
+## Design
+
+### Configuration
+
+#### Label
+
+The client-user should be able to specify where they want the label to be positioned. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Icons
+
+The client-user should be able to specify whether they want left icons on list items or not. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Groups
+
+The client-user should be able to create groups for list items. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+> NOTE: NEEDS VISUAL DESIGN
+
+The client should be able to specify whether they want the group name to be visible on the component when the dropdown is closed.
+
+![ ](spec-images/combobox-26.png)
```

Comment:
Unlike a select, the group of a combobox is not associated with a value. A selection is a unique selection of an item in a list. A combobox is a text input where you use the list of items to autocomplete the text. In a combobox there is no "selection", just autocompleting of some text.

For example, the options:
```
Texas
- Apple 
- Orange
Florida
- Cranberry
- Orange
```

For a select the option of orange under texas and florida are unique and different they may associate to unique numbers 2 and 4 for example that can be differentiated.
For a combobox clicking either option autocompletes the text as "Orange" there is no association to the group, just the text "Orange". You can't tell what group the text was autocompleted from when clicked.

Reply:
I thought clients were specifically looking for this feature in the combo box? To be clear, you're saying the combobox menu can show groups, but the component itself cannot display the group of the current selection?

Reply:
As implemented (by FAST) @rajsite is correct in that there would be no good way for the combobox to display the group correctly when a user selects a value. All the combobox knows is that it has a specific string as its value.

Reply:
Alright, I've removed this specific text.